src/CodeGen_Xtensa.cpp

```diff
@@ -88,8 +91,8 @@ extern int halide_release_dma();
 
 class ScopedDmaInitializer {
  public:
-  ScopedDmaInitializer() {
-    int status = halide_init_dma();
+  ScopedDmaInitializer(int channel_count) {
+    int status = halide_init_dma(channel_count);
```

Comment:
...shouldn't we do something if there's an error? Just ignoring it seems dangerous.

Reply:
Added is_valid() method and corresponding error propogation.